### PR TITLE
feat: add dockerfile

### PR DESCRIPTION
**Pull request by noahlias**

### CODE REVIEW

File: .dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.10.0
+
+LABEL maintainer="Gorgeous-Ghouls"
+
+RUN pip install poetry
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
RUN apt-get -y update
RUN apt-get -y install git curl
RUN curl -sSL https://install.python-poetry.org | python3 -
```
as installing with pip is deprecated

File: .dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN git clone https://github.com/Gorgeous-Ghouls/Blak.git /home/Blak
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
RUN mkdir /server
WORKDIR /server
COPY . .
```
as the `working_directory` in GH action will be `server/`

File: .dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+WORKDIR /home/Blak/server
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
```
no need anymore

File: .dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+RUN poetry install
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
ENV PYTHONPATH=${PYTHONPATH}:${PWD}
ENV PATH="/root/.local/bin:${PATH}"

RUN poetry config virtualenvs.create false
RUN poetry install --no-dev
```
no need to make a venv as we already in docker
same with dev deps

File: .dockerfile [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+EXPOSE 8000
+
+CMD ["poetry", "run","blak-server"]
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
```
i was thinking of using a docker-compose for this

File: .dockerfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.10.0
```
Comment (p0lygun) [inline .dockerfile]:
```suggestion
FROM python:3.10.5-slim-buster
```
latest stable py version

**Review by p0lygun — DISMISSED** [review]
@noahlias  pls update

**Review by p0lygun — APPROVED** [review]
add workflow file

**Review by p0lygun — CHANGES_REQUESTED** [review]
update review

File: .github/workflows/docker-publish.yaml [hunk 7]
```diff
@@ -25,14 +25,13 @@ jobs:
...
 
-
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: true
 
       - name: Build image
-        run: docker build . --file dockerfile --tag $IMAGE_NAME
+        run: docker build . --file .dockerfile --tag $IMAGE_NAME
```
Comment (p0lygun) [inline .github/workflows/docker-publish.yaml]: any specific reason to name it `.dockerfile` ?

Comment (p0lygun) [reply .github/workflows/docker-publish.yaml]: it will be hidden in `ls` is that what u want ?

**Review by p0lygun — COMMENTED** [review]
just one question

File: .github/workflows/docker-publish.yaml (hunk 7 above)
Comment (noahlias) [reply .github/workflows/docker-publish.yaml]: nothing,I will change the name

**Comment by p0lygun** [comment]
ok pls also update 
https://github.com/Gorgeous-Ghouls/Blak/blob/1fe42851e9924fb4bbe750a2b23636bdeab63da5/server/pyproject.toml#L2
to be `blak-server`

**Review by p0lygun — APPROVED** [review]
LGTM !!! good job 👍

**Pull request merged**